mysql-test/main/path_resolution.result

```diff
@@ -0,0 +1,3006 @@
+PROCEDURE proc() AS
+BEGIN
+SELECT 'Invoke from Db: test, Pack: pack, Rt: proc';
+END;
+END;
+$$
+USE pack;
+CREATE PROCEDURE proc() AS
+BEGIN
+SELECT 'Invoke from Db: pack, Rt: proc';
+END;
+$$
+USE test;
+CALL pack.proc;
+Invoke from Db: test, Pack: pack, Rt: proc
+Invoke from Db: test, Pack: pack, Rt: proc
+DROP PACKAGE BODY pack;
+DROP PACKAGE pack;
+CALL pack.proc;
+Invoke from Db: pack, Rt: proc
+Invoke from Db: pack, Rt: proc
+DROP PROCEDURE pack.proc;
+CALL pack.proc;
+ERROR 42000: PROCEDURE pack.proc does not exist
+SET @@path = DEFAULT;
+#
+# type 8.13
+# sql_mode: ORACLE | procedure | is qualified
+# curr db: test2 | path set: yes
+# resolution priority in between qualified package procedure vs qualified db procedure resolution
+#
+USE test;
+CREATE PACKAGE pack AS
+PROCEDURE proc();
+END;
+$$
+CREATE PACKAGE BODY pack AS
+PROCEDURE proc() AS
+BEGIN
+SELECT 'Invoke from Db: test, Pack: pack, Rt: proc';
+END;
+END;
+$$
+USE pack;
+CREATE PROCEDURE proc() AS
+BEGIN
+SELECT 'Invoke from Db: pack, Rt: proc';
+END;
+$$
+USE test2;
+CALL pack.proc;
+Invoke from Db: pack, Rt: proc
+Invoke from Db: pack, Rt: proc
+SET PATH= 'sys,test';
+CALL pack.proc;
+Invoke from Db: test, Pack: pack, Rt: proc
+Invoke from Db: test, Pack: pack, Rt: proc
+DROP PACKAGE BODY test.pack;
+DROP PACKAGE test.pack;
+CALL pack.proc;
+Invoke from Db: pack, Rt: proc
+Invoke from Db: pack, Rt: proc
+DROP PROCEDURE pack.proc;
+CALL pack.proc;
+ERROR 42000: PROCEDURE pack.proc does not exist
+DROP DATABASE pack;
+SET @@path = DEFAULT;
+#
+# type 8.14
+# sql_mode: ORACLE | procedure | is qualified
+# curr db: X | path set: yes
+# qualified package procedure invoke internal private procedure
+#
+SET PATH= 'test,test2';
+USE test2;
+CREATE PROCEDURE proc_prv() AS
+BEGIN
+SELECT 'Invoke from Db: test2, Rt: proc_prv';
+END;
+$$
+USE test;
+CREATE PROCEDURE proc_prv() AS
+BEGIN
+SELECT 'Invoke from Db: test, Rt: proc_prv';
+END;
+$$
+CREATE PACKAGE pack AS
+PROCEDURE proc_pub();
+END;
+$$
+CREATE PACKAGE BODY pack AS
+PROCEDURE proc_prv() AS
+BEGIN
+SELECT 'Invoke from Db: test, Pack: pack, Rt: proc_prv';
+END;
+PROCEDURE proc_pub() AS
+BEGIN
+CALL proc_prv;
+END;
+END;
+$$
+CREATE DATABASE deselect_db;
+USE deselect_db;
+DROP DATABASE deselect_db;
+CALL pack.proc_pub;
+Invoke from Db: test, Pack: pack, Rt: proc_prv
+Invoke from Db: test, Pack: pack, Rt: proc_prv
+SET @@path = DEFAULT;
+#
+# type 8.15
+# sql_mode: ORACLE | procedure | is qualified
+# curr db: test2 | path set: yes
+# qualified package procedure invoke internal private procedure
+#
+SET PATH= 'test,test2';
+USE test2;
+CALL pack.proc_pub;
+Invoke from Db: test, Pack: pack, Rt: proc_prv
+Invoke from Db: test, Pack: pack, Rt: proc_prv
+DROP PROCEDURE proc_prv;
+DROP DATABASE test2;
+USE test;
+DROP PROCEDURE proc_prv;
+DROP PACKAGE BODY pack;
+DROP PACKAGE pack;
+SET @@path = DEFAULT;
+#
+# Resolution of routines in stored routines should be based on
+# the current path, not the path at the time of creation.
```

Comment:
No, quite the opposite. As I already commented above, should be based on the path at the time of the creation.